### DISCUSSION TITLE
Exit campaign early

### DISCUSSION THREAD

**Discussion by jordanjoyce** [parcelvoy/platform#361] [Ideas]
Would love the ability to end journeys early, on a user by user basis, via either the API or when an event occurs for a user. We have some long running flows, and if we get a conversion in that time frame it would be great to be able to end the journey for the user.

**Comment by williamhrs** [comment]
We would also love to have that ability, we send lots of e-mails and we have to remove people that don't open them anymore from a running journey

**Reply by williamhrs** [reply]
@pushchris is there a way to do that?

**Reply by pushchris** [reply]
You can remove a user from a journey via API (https://github.com/parcelvoy/platform/issues/407) but not currently from the journey UI. We are starting to explore how this might work in #501 but nothing official yet

**Comment by pushchris** [comment]
Exit criteria for journeys is now available in our latest [release](https://github.com/parcelvoy/platform/releases/tag/v1.9.0), would love for you to give it a whirl!

**Reply by williamhrs** [reply]
will update and try it out !

**Comment by leobarcellos** [comment]
@pushchris If we leave the step like this, without any entrances, will it exit from all journeys? (If not, is it possible to work like this?)

Or maybe, make the entrance selector an array input and allow to select entrances from other journeys too, what do you think?

![image](https://github.com/user-attachments/assets/4c6e1b8e-8720-4b77-b428-ed9a2efe6c7a)

**Reply by pushchris** [reply]
Looks like you have two issues here:
1. There is no entrance to that exit so it'll never be reached, you need to define some criteria for when this step will happen and how a user will get to it
2. No, it currently only allows you to pick entrances in the same journey and only one. Can explore allowing for additional selections, but in general each journey should have criteria to be met for exiting.

**Reply by leobarcellos** [reply]
@pushchris 
Oh, the screenshot was just an example. I can see the entrance just fine, however we do have several entrances on multiple journeys, and we need to remove the user from all entrances (we did some journeys using events and each one have an entrance)

The way it's implemented right now, we would need to have several exit steps for each entrances, it's going to be messy.
Maybe allow to choose "all" or select several entrances on the exit step would be good.
Or maybe another step that just remove the user.
